src/App.tsx

```diff
@@ -1,61 +1,157 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import Box from '@mui/material/Box';
+import { Button, MenuItem } from '@mui/material';
+import TextField from '@mui/material/TextField';
+import FormHelperText from '@mui/material/FormHelperText';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function findUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const preparedTodos: Todo[] = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: findUserById(todo.userId),
+  };
+});
+
+function findUserByName(userName: string): User | null {
+  return usersFromServer.find(user => user.name === userName) || null;
+}
 
 export const App = () => {
+  const defaultUserOption = 'Choose a user';
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(defaultUserOption);
+  const [todos, setTodos] = useState(preparedTodos);
+  const [titleErrorAlert, setTitleErrorAlert] = useState(false);
+  const [selectedUserErrorAlert, setSelectedUserErrorAlert] = useState(false);
+
+  const handleSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (selectedUser !== defaultUserOption && title !== '') {
+      const newUser = findUserByName(selectedUser);
+
+      setTodos(currentTodo => {
+        const maxTodoId = Math.max(...currentTodo.map(todo => todo.id));
+
+        setTitle('');
+        setSelectedUser(defaultUserOption);
+
+        return [
+          ...currentTodo,
+          {
+            id: maxTodoId + 1,
+            title,
+            completed: false,
+            userId: newUser ? newUser.id : null,
+            user: newUser,
+          },
+        ];
+      });
+    }
+
+    if (selectedUser === defaultUserOption) {
+      setSelectedUserErrorAlert(true);
+    }
+
+    if (title === '') {
+      setTitleErrorAlert(true);
+    }
```

Comment:
Let's move this condition to the top of the function and add a return as we did in the lesson. Then remove `if` in add todo block. It improvements increase code readable